hvantk/algorithms/ptm/constraint.py

```
@@ -300,15 +336,107 @@ def _load_variants(config: PTMConstraintConfig) -> pd.DataFrame:
+            )
+
+    return out
+
+
+def _read_variants_tabular_file(path: str) -> pd.DataFrame:
+    """Read a variant table from pickle/parquet/CSV/TSV (gzip/bgz transparent)."""
+    from pathlib import Path
+
+    suffixes = [s.lower() for s in Path(path).suffixes]
+    if ".pkl" in suffixes or ".pickle" in suffixes:
+        return pd.read_pickle(path)
```

Ruff (0.15.15)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)